src/mantine-core/src/components/Notification/Notification.module.css

```diff
@@ -0,0 +1,156 @@
+.root {
+  overflow: hidden;
+  box-sizing: border-box;
+  position: relative;
+  display: flex;
+  align-items: center;
+  padding-left: var(--_root-padding-left, rem(22px));
+  padding-right: var(--_root-padding-right, var(--mantine-spacing-xs));
+  padding-top: var(--mantine-spacing-xs);
+  padding-bottom: var(--mantine-spacing-xs);
+  border-radius: var(--notification-radius);
+  box-shadow: var(--mantine-shadow-lg);
+  background-color: var(--_bg);
+
+  &::before {
+    content: '';
+    display: block;
+    position: absolute;
+    width: rem(6px);
+    top: var(--notification-radius);
+    bottom: var(--notification-radius);
+    left: rem(4px);
+    border-radius: var(--notification-radius);
+    background-color: var(--notification-color);
+
+    @mixin rtl {
+      right: rem(4px);
+      left: auto;
+    }
+  }
+
+  &[data-with-icon] {
+    --_root-padding-left: var(--mantine-spacing-xs);
+
+    &::before {
+      display: none;
+    }
+
+    @mixin rtl {
+      --_root-padding-right: var(--mantine-spacing-xs);
+    }
+  }
+
+  &[data-with-border] {
+    border: rem(1px) solid var(--_border-color);
```

Comment:
Do not use CSS properties in `data-` attributes, it makes it impossible to override styles, define border on the element and change variable in data- attribute.